src/storage/mvcc/mvcc_read_tracker.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright 2025 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    sync::{
+        Arc,
+        atomic::{AtomicBool, AtomicU64, Ordering},
+    },
+    thread,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use dashmap::DashMap;
+use lazy_static::lazy_static;
+use tikv_util::sys::thread::StdThreadBuildWrapper;
+
+/// Default time window for MVCC read tracking (in seconds)
+const DEFAULT_MVCC_READ_WINDOW_SECS: u64 = 300; // same as compaction check interval
+
+lazy_static! {
+    pub static ref MVCC_READ_TRACKER: MvccReadTracker = MvccReadTracker::new();
+}
+
+/// Simple atomic counter for tracking MVCC versions scanned per region
+#[derive(Debug)]
+pub struct RegionMvccReadStats {
+    /// Total MVCC versions scanned (atomic for lock-free updates)
+    total_mvcc_versions_scanned: AtomicU64,
+    /// Total number of read requests (atomic for lock-free updates)
+    total_requests: AtomicU64,
+    /// Timestamp (in seconds since epoch) of the last read
+    last_read_time_secs: AtomicU64,
+}
+
+impl RegionMvccReadStats {
+    pub fn new() -> Self {
+        Self {
+            total_mvcc_versions_scanned: AtomicU64::new(0),
+            total_requests: AtomicU64::new(0),
+            last_read_time_secs: AtomicU64::new(0),
+        }
+    }
+
+    /// Add MVCC versions scanned and increment request counter (lock-free
+    /// atomic increment)
+    fn add_mvcc_versions(&self, mvcc_versions: u64) {
+        self.total_mvcc_versions_scanned
+            .fetch_add(mvcc_versions, Ordering::Relaxed);
+        self.total_requests.fetch_add(1, Ordering::Relaxed);
+    }
+
+    /// Get total MVCC versions scanned in current window
+    pub fn get_total_mvcc_versions(&self) -> u64 {
+        self.total_mvcc_versions_scanned.load(Ordering::Relaxed)
+    }
+
+    /// Get total number of requests in current window
+    pub fn get_total_requests(&self) -> u64 {
+        self.total_requests.load(Ordering::Relaxed)
+    }
+
+    /// Get timestamp of last read (in seconds since epoch)
+    pub fn get_last_read_time_secs(&self) -> u64 {
+        self.last_read_time_secs.load(Ordering::Relaxed)
+    }
+}
+
+impl Default for RegionMvccReadStats {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Global tracker for MVCC read activity across all regions (lock-free with
+/// DashMap)
+#[derive(Clone)]
+pub struct MvccReadTracker {
+    /// Lock-free concurrent map from region_id to read statistics
+    stats: Arc<DashMap<u64, RegionMvccReadStats>>,
+    /// Flag to stop background reset thread
+    stop_flag: Arc<AtomicBool>,
+    /// Time window in seconds
+    window_secs: u64,
```

Comment:
what if this window is configured to a value that is proportional to compaction window, then every time compaction runner checks for candidates, it sees empty stats. Is there a better way to avoid such problem? like sliding window?